test/test_image.py

```diff
@@ -0,0 +1,473 @@
+
+import pytest
+
+from docker_cli.exceptions import CommandNotFoundOnImageError
+from docker_cli.image import Image, get_image_id
+
+
+@pytest.fixture
+def test_image_id():
+    """
+    Builds an image for testing and returns its ID.
+    """
+    run(split("docker build ./ -t test"))
+    inspect_process = run(
+        split("docker inspect -f='{{.Id}}' test"),
+        capture_output=True,
+        text=True,
+        check=True,
+    )
+    return inspect_process.stdout.strip()
+
+
+def test_init(test_image_id):
+    """
+    Tests that the __init__ function on the Image class is correctly
+    receiving and remembering the ID of a docker image.
+    """
+    id = test_image_id
+    print("ID: " + id)
+    img = Image("test")
+
+    assert img is not None
+    assert img.id == id
+
+    img_2 = Image(id)
+    assert img_2.id == id
+
+    run(split("docker image remove test"))
+
+
+def test_init_CalledProcessError():
+    """
+    Tests that the __init__ function on the Image class raises a
+    CalledProcessError when given a malformed name."""
+    img = None
+    with pytest.raises(CalledProcessError):
+        img = Image("malformed_image_name_or_id")
+    assert img is None
+
+
+def test_build_from_dockerfile():
+    """
+    Tests that the build method constructs and returns an Imagemage when
+    given a Dockerfile.
+    """
+    img = Image.build(tag="test")
+    inspect_process = run(
+        split("docker inspect -f='{{.Id}}' test"),
+        text=True,
+        capture_output=True)
+    id = inspect_process.stdout.strip()
+
+    assert img is not None
+    assert img.id == id
+    run(split("docker image remove test"))
+
+
+def test_build_from_dockerfile_output_to_file():
+    """
+    Tests that the build method writes to a file when configured to
+    do so.
+    """
+    file = open("testfile.txt", "w")
+    img = Image.build(
+        tag="test",
+        dockerfile="./Dockerfile",
+        stdout=file,
+        stderr=file
+    )
+    inspect_process = run(
+        split("docker inspect -f='{{.Id}}' test"),
+        text=True,
+        capture_output=True)
+    id = inspect_process.stdout.strip()
+    file.close()
+
+    assert img is not None
+    assert img.id == id
+    assert os.stat("testfile.txt").st_size > 0
+
+    os.remove("testfile.txt")
+    run(split("docker image remove test"))
+
+
+def test_build_from_dockerfile_dockerfile_in_different_location():
+    """
+    Tests that the build method can build an image from a dockerfile in a
+    different location than the context root directory.
+    """
+    img = Image.build(
+        tag="test", dockerfile="dockerfiles/alpine_functional.dockerfile"
+    )
+    inspect_process = run(
+        split("docker inspect -f='{{.Id}}' test"),
+        text=True,
+        capture_output=True)
+    id = inspect_process.stdout.strip()
+
+    assert img is not None
+    assert img.id == id
+    run(split("docker image remove test"))
```

Comment:
Let's wrap this test body (as well as the other similar tests) in a `try`/`finally` block, like this:

```suggestion
    try:
        img = Image.build(
            tag="test", dockerfile="dockerfiles/alpine_functional.dockerfile"
        )
        inspect_process = run(
            split("docker inspect -f='{{.Id}}' test"),
            text=True,
            capture_output=True)
        id = inspect_process.stdout.strip()
    
        assert img is not None
        assert img.id == id
    finally:
        run(split("docker image remove test"))
```

This will ensure that the `docker image remove` command is always run at the end to cleanup even if the test assertions fail or an exception is raised in the test body.